rp235x-hal-examples/Cargo.toml

```diff
@@ -28,10 +28,15 @@ futures = {version = "0.3.30", default-features = false, features = ["async-awai
 hd44780-driver = "0.4.0"
 heapless = "0.8.0"
 nb = "1.0"
-nostd_async = {version = "0.6.1", features = ["cortex_m"]}
 panic-halt = "0.2.0"
 pio = "0.2.0"
 pio-proc = "0.2.0"
 rp235x-hal = {path = "../rp235x-hal", version = "0.2.0", features = ["binary-info", "critical-section-impl", "rt", "defmt"]}
 usb-device = "0.3.2"
 usbd-serial = "0.2.2"
+
+[target.'thumbv8m.main-none-eabihf'.dependencies]
+nostd_async = {version = "0.6.1", features = ["cortex_m"]}
+
+[target.riscv32imac-unknown-none-elf.dependencies]
+nostd_async = {version = "0.6.1"}
```

Comment:
I don't think `nostd_async` should be used without the `cortex_m` feature. The fallback implementation of `interrupt::free` just doesn't disable any interrupts and unsafely creates a `CriticalSection` token anyway:

https://github.com/sammhicks/nostd_async/blob/master/src/interrupt.rs#L5-L10

Reply:
Yeah. Not sure what we should do instead though.

Reply:
I'd say either embassy-executor (AFAIKT the most popular no_std executor), or some minimal toy executor like the one I suggested for on-target tests in #848.

There are two conflicting goals when choosing the executor to be used for the examples:
- Examples should be concise
- Examples should introduce knowledge that's applicable to real-world tasks

Depending on what you value more, you could justify both choices.
In any case, nostd_async IMHO isn't a good choice, as it may look like it's usable for real (non-example) code, while in fact it has severe issues.

I'm not knowledgeable enough in writing async executors to decide if a third option could be viable: Improving nostd_async. It may only need a few small fixes, or it may be fundamentally flawed - I just don't know.

Reply:
https://github.com/rp-rs/rp-hal/pull/850